src/App.tsx

```diff
@@ -1,14 +1,49 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState('all');
+  const [query, setQuery] = useState('');
+  const [load, setLoad] = useState<Todo[]>(todos);
+
+  const lowerCompare = (str: string) => (
+    str.toLowerCase().includes(query.toLowerCase())
+  );
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  useEffect((() => {
+    switch (filterBy) {
```

Comment:
To make the code more understandable, better create a separate variable  for filtered todos
for example:
```suggestion
   const filteredTodos = todos
    .filter(({ destructure }) => {
      switch (filterBy) {
        //code
      }
    });
```
and then `setLoad(filteredTodos)`.